### PR TITLE
[Fix] Transformation Matrix Mis-calculation for autoaugmentations

### LINKED ISSUE #2843: RanAug test failing
### Describe the bug

The test `test_transform_mat` in RandomAugmenter is  failing 
https://github.com/kornia/kornia/blob/main/tests/augmentation/test_auto_operation.py#L119

It seems to be because the operation `get_transformation_matrix` (from parameters) changes the stage of the `aug.transform_matrix`

This version of the test does not fail
image (51)

Whereas this one, which is current version, does 
image (50)

### Reproduction steps
Just run the tests

```bash
pytest -v --device cpu --dtype float64 tests/augmentation/test_auto_operation.py 
```

And you will see:

```
tests/augmentation/test_auto_operation.py:152: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
testing/base.py:106: in assert_close
    return assert_close(actual, expected, rtol=rtol, atol=atol)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

actual = tensor([[[ 1.0000,  0.0000,  0.0000],
         [ 0.0000,  1.0000, 13.4930],
         [ 0.0000,  0.0000,  1.0000]],

  ...,  0.0000,  0.0000],
         [ 0.0000,  1.0000, 16.0136],
         [ 0.0000,  0.0000,  1.0000]]], dtype=torch.float64)
expected = tensor([[[ 1.0000,  0.0000,  0.0000],
         [ 0.0000,  1.0000, 17.1878],
         [ 0.0000,  0.0000,  1.0000]],

  ...,  0.0000,  0.0000],
         [ 0.0000,  1.0000, 10.7947],
         [ 0.0000,  0.0000,  1.0000]]], dtype=torch.float64)
rtol = 1e-05, atol = 1e-05, kwargs = {}

    def assert_close(
        actual: Tensor, expected: Tensor, *, rtol: Optional[float] = None, atol: Optional[float] = None, **kwargs: Any
    ) -> None:
        if rtol is None and atol is None:
            # `torch.testing.assert_close` used different default tolerances than `torch.testing.assert_allclose`.
            # TODO: remove this special handling as soon as https://github.com/kornia/kornia/issues/1134 is resolved
            #  Basically, this whole wrapper function can be removed and `torch.testing.assert_close` can be used
            #  directly.
            rtol, atol = _default_tolerances(actual, expected)
    
>       return _assert_close(
            actual,
            expected,
            rtol=rtol,
            atol=atol,
            # this is the default value for torch>=1.10, but not for torch==1.9
            # TODO: remove this if kornia relies on torch>=1.10
            check_stride=False,
            equal_nan=False,
            **kwargs,
        )
E       AssertionError: Tensor-likes are not close!
E       
E       Mismatched elements: 10 / 90 (11.1%)
E       Greatest absolute difference: 11.92533016204834 at index (7, 1, 2) (up to 1e-05 allowed)
E       Greatest relative difference: 1.889444158005776 at index (2, 1, 2) (up to 1e-05 allowed)

testing/base.py:37: AssertionError
....

FAILED tests/augmentation/test_auto_operation.py::TestTrivialAugment::test_transform_mat[cpu-float64] - AssertionError: Tensor-likes are not close!
```
```


### Expected behavior

not fail 

### Environment

```shell
Collecting environment information...
PyTorch version: 2.2.1+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Debian GNU/Linux 10 (buster) (x86_64)
GCC version: (GCC) 10.4.0
Clang version: Could not collect
CMake version: version 3.25.0
Libc version: glibc-2.28

Python version: 3.8.12 (default, Oct 12 2021, 13:49:34)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-4.19.0-26-cloud-amd64-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: 12.2.128
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: Tesla V100-SXM2-16GB

Nvidia driver version: 535.86.10
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
Address sizes:       46 bits physical, 48 bits virtual
CPU(s):              32
On-line CPU(s) list: 0-31
Thread(s) per core:  2
Core(s) per socket:  16
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               63
Model name:          Intel(R) Xeon(R) CPU @ 2.30GHz
Stepping:            0
CPU MHz:             2299.998
BogoMIPS:            4599.99
Hypervisor vendor:   KVM
Virtualization type: full
L1d cache:           32K
L1i cache:           32K
L2 cache:            256K
L3 cache:            46080K
NUMA node0 CPU(s):   0-31
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm invpcid_single pti ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid xsaveopt arat md_clear arch_capabilities

Versions of relevant libraries:
[pip3] flake8==5.0.4
[pip3] mypy==0.982
[pip3] mypy-extensions==0.4.3
[pip3] natten==0.14.6+torch200cu118
[pip3] numpy==1.24.4
[pip3] numpy-quaternion==2020.9.5.14.42.2
[pip3] onnx==1.14.0
[pip3] onnxruntime==1.14.1
[pip3] open-clip-torch==2.0.2
[pip3] pytest-mypy==0.10.0
[pip3] pytorch-lightning==1.5.0
[pip3] pytorch-metric-learning==2.4.1
[pip3] pytorch-triton==2.1.0+6e4932cda8
[pip3] pytorch3d==0.7.2
[pip3] torch==2.2.1
[pip3] torch-lr-finder==0.2.1
[pip3] torchaudio==2.2.0.dev20231115+cu121
[pip3] torchfile==0.1.0
[pip3] torchgeometry==0.1.2
[pip3] torchmetrics==0.11.4
[pip3] torchsde==0.2.6
[pip3] torchvision==0.17.1
[pip3] triton==2.2.0
[pip3] triton-nightly==2.1.0.dev20230608213938
[conda] natten                    0.14.6+torch200cu118          pypi_0    pypi
[conda] numpy                     1.24.4                   pypi_0    pypi
[conda] numpy-quaternion          2020.9.5.14.42.2          pypi_0    pypi
[conda] open-clip-torch           2.0.2                    pypi_0    pypi
[conda] pytorch-lightning         1.5.0                    pypi_0    pypi
[conda] pytorch-metric-learning   2.4.1                    pypi_0    pypi
[conda] pytorch-triton            2.1.0+6e4932cda8          pypi_0    pypi
[conda] pytorch3d                 0.7.2                    pypi_0    pypi
[conda] torch                     2.2.1                    pypi_0    pypi
[conda] torch-lr-finder           0.2.1                    pypi_0    pypi
[conda] torchaudio                2.2.0.dev20231115+cu121          pypi_0    pypi
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchgeometry             0.1.2                    pypi_0    pypi
[conda] torchmetrics              0.11.4                   pypi_0    pypi
[conda] torchsde                  0.2.6                    pypi_0    pypi
[conda] torchvision               0.17.1                   pypi_0    pypi
[conda] triton                    2.2.0                    pypi_0    pypi
[conda] triton-nightly            2.1.0.dev20230608213938          pypi_0    pypi
```
```


### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shijianjian**
Fixes #2843

### CODE REVIEW

File: tests/augmentation/test_auto_operation.py [hunk 1]
```diff
@@ -112,10 +111,6 @@ def test_smoke(self, policy):
         in_tensor = torch.rand(10, 3, 50, 50, requires_grad=True)
         aug(in_tensor)
 
-    @pytest.mark.xfail(
-        torch_version() in {"1.9.1", "1.10.2", "1.11.0", "1.12.1", "1.13.1"},
-        reason="randomness failing into some torch versions",
```
Comment (johnnv1) [inline tests/augmentation/test_auto_operation.py:117]: did you test it locally? on PR the CI isn't running the older versions

**Review by johnnv1 — APPROVED** [review]

File: tests/augmentation/test_auto_operation.py (hunk 1 above)
Comment (shijianjian) [reply tests/augmentation/test_auto_operation.py:117]: yes. should be fine.

**Comment by shijianjian** [comment]
weird. It passed on my local machine, but not here.

**Comment by johnnv1** [comment]
> weird. It passed on my local machine, but not here.

I'm not sure if it's me who added the skip, but if it was, that's why it was like randomness 😢 

maybe it's just an adjust in the tolerance?

**Review by edgarriba — APPROVED** [review]

**Comment by shijianjian** [comment]
> > weird. It passed on my local machine, but not here.
> 
> I'm not sure if it's me who added the skip, but if it was, that's why it was like randomness 😢
> 
> maybe it's just an adjust in the tolerance?

It is added by Dmytro I think. The point is after my fix, it totally works on my machine under Python3.8 and torch1.9.1. I do not know what is going on in the CI. XD

**Comment by johnnv1** [comment]
> It is added by Dmytro I think. The point is after my fix, it totally works on my machine under Python3.8 and torch1.9.1. I do not know what is going on in the CI. XD

in my local it is randomly falling xD

```bash
$ conda create -p venv38 python=3.8
$ conda install pytorch=1.9.1 cpuonly -c pytorch -c nvidia
$ pip install accelerate==0.20.3 # otherwise will install new torch
$ pip install -e .[dev,x]
# run it until fail
$ pytest tests/augmentation/test_auto_operation.py --device=cpu --dtype=float64,float32 -vv
```

some times the `AILED tests/augmentation/test_auto_operation.py::TestOperations::test_step_routine[op5]` also fail with 
```output
E       assert 0.5 != 0.5
E        +  where 0.5 = <built-in method item of Tensor object at 0x7ae54b77a900>()
E        +    where <built-in method item of Tensor object at 0x7ae54b77a900> = tensor([0.5000], grad_fn=<ClampBackward1>).item
E        +      where tensor([0.5000], grad_fn=<ClampBackward1>) = HorizontalFlip(\n  (op): RandomHorizontalFlip(p=0.5, p_batch=1.0, same_on_batch=False)\n).probability
```

btw, the number of warnings also changes a lot between each run

**Pull request merged**